packages/teams-js/src/internal/validDomains.ts

```diff
@@ -0,0 +1,104 @@
+import { GlobalVars } from '../internal/globalVars';
+import { validDomainsCdnEndpoint, validOriginsFallback } from './constants';
+import { getLogger } from './telemetry';
+import { isValidHttpsURL } from './utils';
+
+let validOrigins: string[] = [];
+export async function prefetchDomainsFromCDN(): Promise<void> {
+  if (fetch) {
+    await fetch(validDomainsCdnEndpoint);
+  }
+}
+
+async function retrieveDomainsFromCDNAndStore(): Promise<string[]> {
+  if (validOrigins.length !== 0) {
+    return validOrigins;
+  }
+  if (fetch) {
+    return fetch(validDomainsCdnEndpoint)
+      .then((response) => {
+        if (!response.ok) {
+          validOrigins = validOriginsFallback;
+          return validOriginsFallback;
+        }
+        return response.json().then((validDomains) => {
+          validOrigins = validDomains.validOrigins;
```

Comment:
>validDomains.validOrigins [](http://example.com/codeflow?start=25&length=25)

I have to say that the "sometimes use domains" and "sometimes use origins" naming has tripped me up a few times.

Reply:
rewrote it to just use some variation of validOrigins to clarify things, hopefully that helps